web-src/src/helpers/PromptRenderer.js

```diff
@@ -54,7 +54,31 @@ export function createContentModelPrompt(contentFragmentModel) {
     + '\n```';
 }
 
-export function renderPrompt(prompt, placeholders, contentFragmentModel) {
-  const extraPrompt = contentFragmentModel ? createContentModelPrompt(contentFragmentModel) : '';
-  return removeEmptyLines(resolvePlaceholders(prompt, placeholders)) + extraPrompt;
+export function addContentFragmentContext(fragment) {
+  if (!fragment) return '';
+
+  return '\n\nOriginal content fragment for context: ```'
+  + `\n${fragment.fields
+    .map((field) => {
+      return (field.values.length > 0 ? `- ${field.name}: ${field.values}\n` : '');
+    })
+    .join('')}`
+  + '```'
+  + '\n\nContent fragment variations for context: ```'
```

Comment:
I wouldn't add it since most variations, except the original one, will likely be autogenerated. We shouldn't condition the model's output on its own input.